### PR TITLE
feat(conf): export hostgroup and servicegroup as tags

### PR DESCRIPTION
**Pull request by kduret**
## Description

export hostgroup and servicegroup as tags

**Fixes** MON-13050

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

### CODE REVIEW

File: www/class/config-generate/servicegroup.class.php [hunk 1]
```diff
@@ -36,26 +36,20 @@
...
+    private const TAG_TYPE = 'servicegroup';
+
     private $use_cache = 1;
     private $done_cache = 0;
 
     private $sg = array();
-    private $sg_relation_cache = array();
     protected $generate_filename = 'servicegroups.cfg';
     protected $object_name = 'servicegroup';
+    private $sg_relation_cache = array();
```
Comment (a-launois) [inline www/class/config-generate/servicegroup.class.php:47]:
```suggestion
    private $sg_relation_cache = [];
```

File: www/class/config-generate/hostgroup.class.php [hunk 2]
```diff
@@ -92,8 +83,37 @@ public function addHostInHg($hg_id, $host_id, $host_name)
...
+        $this->generateTags();
+    }
+
+    /**
+     * Generate host groups and write in file
+     */
+    private function generateHostGroups(): void
+    {
+        $this->generate_filename = 'hostgroups.cfg';
+        $this->object_name = 'hostgroup';
```
Comment (kduret) [inline www/class/config-generate/hostgroup.class.php:101]: use constants

File: www/class/config-generate/hostgroup.class.php [hunk 3]
```diff
@@ -104,6 +124,38 @@ public function generateObjects()
         }
     }
 
+    /**
+     * Generate tags and write in file
+     */
+    private function generateTags(): void
+    {
+        $this->generate_filename = 'tags.cfg';
+        $this->object_name = 'tag';
```
Comment (kduret) [inline www/class/config-generate/hostgroup.class.php:133]: use constants

File: www/class/config-generate/servicegroup.class.php (hunk 1 above)
Comment (kduret) [inline www/class/config-generate/servicegroup.class.php:47]: retrieve previous change

File: www/class/config-generate/servicegroup.class.php [hunk 4]
```diff
@@ -193,6 +213,38 @@ public function generateObjects()
         }
     }
 
+    /**
+     * Generate tags and write in file
+     */
+    private function generateTags(): void
+    {
+        $this->generate_filename = 'tags.cfg';
+        $this->object_name = 'tag';
```
Comment (kduret) [inline www/class/config-generate/servicegroup.class.php:222]: use constants

**Review by adr-mo — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**